cmd/clusterctl/cmd/root.go

```diff
@@ -46,6 +48,15 @@ var RootCmd = &cobra.Command{
 	Long: LongDesc(`
 		Get started with Cluster API using clusterctl to create a management cluster,
 		install providers, and create templates for your workload cluster.`),
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// Check if Config folder (~/.clusterctl) exist and if not create it
+		if _, err := os.Stat(config.ConfigFolder); os.IsNotExist(err) {
+			if err := os.MkdirAll(filepath.Dir(config.ConfigFolder), os.ModePerm); err != nil {
```

Comment:
I wonder if it would be better to explicitly include the `homedir.HomeDir()` when making the directory. I mention this because not sure of the behavior on other OSes like Windows.
Also we are doing this explicit path when trying to delete the downloaded file.

Reply:
missed that, thanks for the catch! and thanks for your review!